tests/accelerators/test_common.py

```diff
@@ -24,6 +31,7 @@ def test_auto_device_count(device_count_mock):
     assert GPUAccelerator.auto_device_count() == 2
     assert TPUAccelerator.auto_device_count() == 8
     assert IPUAccelerator.auto_device_count() == 4
+    assert HPUAccelerator.auto_device_count() == 8
```

Comment:
there's no reason this has to be in `test_common`. testing the HPU Accelerator's auto_device_count implementation should live in `accelerators/test_hpu.py` and we should do the same for the other lines in this test (in separate PRs).